lib/cryptoexchange/client.rb

```diff
@@ -24,6 +26,30 @@ def ticker(market_pair)
           t.base == market_pair.base && t.target == market_pair.target
         end
       end
+    rescue HTTP::ConnectionError, HTTP::TimeoutError, JSON::ParserError
+      return { error: "#{exchange}'s service is temporarily unavailable." }
+    end
+
+    def available_exchanges
+      folder_names = Dir[File.join(File.dirname(__dir__), 'cryptoexchange', 'exchanges', '**')]
+      folder_names.map do |folder_name|
+        folder_name.split('/').last
+      end
+    end
+
+    def exchange_for(currency)
+      exchanges = []
+      available_exchanges.each do |exchange|
+        pairs = pairs(exchange)
+        next if pairs.is_a?(Hash) && !pairs[:error].empty?
+        pairs.each do |pair|
```

Comment:
@Swingcloud you are still going to have duplicates, can you call `uniq` in the end?

Reply:
also, would be great if you can document this method in README

Reply:
seems like if it has an equal value on base or target, that will break and go to the other exchange.

and I'll document it